lib/ansible/modules/cloud/huawei/hwc_vpc_route.py

```diff
@@ -0,0 +1,422 @@
+      - "name"
+    peering_vpc:
+      vpc_id: "{{ vpc2.id }}"
+  register: connect
+- name: create a route
+  hwc_vpc_route:
+    vpc_id: "{{ vpc1.id }}"
+    destination: "192.168.0.0/16"
+    next_hop: "{{ connect.id }}"
+'''
+
+RETURN = '''
+    id:
+        description:
+            - UUID of the route.
+        type: str
+        returned: success
+    destination:
+        description:
+            - Specifies the destination IP address or CIDR block.
+        type: str
+        returned: success
+    next_hop:
+        description:
+            - Specifies the next hop. The value is VPC peering connection ID.
+        type: str
+        returned: success
+    vpc_id:
+        description:
+            - Specifies the VPC ID to which route is added.
+        type: str
+        returned: success
+    type:
+        description:
+            - Specifies the type of route.
+        type: str
+        returned: success
+'''
+
+from ansible.module_utils.hwc_utils import (
+    Config, HwcClientException, HwcModule, are_different_dicts, build_path,
+    get_region, is_empty_value, navigate_value)
+
+
+def build_module():
+    return HwcModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'],
+                       type='str'),
+            destination=dict(type='str', required=True),
+            next_hop=dict(type='str', required=True),
+            vpc_id=dict(type='str', required=True),
+            type=dict(type='str', default='peering')
+        ),
+        supports_check_mode=True,
+    )
+
+
+def main():
+    """Main function"""
+
+    module = build_module()
+    config = Config(module, "vpc")
+
+    try:
+        resource = None
+        v = search_resource(config)
+        if len(v) > 1:
+            raise Exception("find more than one resources(%s)" % ", ".join([
+                            navigate_value(i, ["id"]) for i in v]))
+
+        if len(v) == 1:
+            resource = v[0]
+            module.params['id'] = navigate_value(resource, ["id"])
+
+        result = {}
+        changed = False
+        if module.params['state'] == 'present':
+            if resource is None:
+                if not module.check_mode:
+                    create(config)
+                changed = True
+
+            result = read_resource(config)
+            result['id'] = module.params.get('id')
+        else:
+            if resource:
+                if not module.check_mode:
+                    delete(config)
+                changed = True
+
+    except Exception as ex:
+        module.fail_json(msg=str(ex))
+
+    else:
+        result['changed'] = changed
+        module.exit_json(**result)
+
+
+def user_input_parameters(module):
+    return {
+        "destination": module.params.get("destination"),
+        "next_hop": module.params.get("next_hop"),
+        "type": module.params.get("type"),
+        "vpc_id": module.params.get("vpc_id"),
+        "id": module.params.get("id"),
```

Comment:
```suggestion
        "id": None,
```
Since `id` is no longer a module option, it will always give `None`.

Reply:
Added id as the input paramters